### ISSUE TITLE
Company level results - Issue remains 

### ISSUE THREAD

**Issue by georgeharris2deg** [2DegreesInvesting/r2dii.analysis#255]
The company-level results Issue whereby unweighted results are coming out wrong when there are multiple loans to the same company is still an issue. 

I believe Jackson had fixed this via the most recent hotfix on CRAN - however, one user still reports the same error. 

What the code should do is simply report the projected production values for any given company directly from the ALD. 
eg 
target_market_share(data, ald, scenario, region, by_company = TRUE, weighted_production = FALSE) 

An additional issue which may or may not be related is that in the output data frame of the above function there is a "corporate_economy" appearing in the column "scenario_source" which is odd as the corporate economy should only appear in the column "metric"

@maurolepore

**Comment by maurolepore** [comment]
* Context: @georgeharris2deg reported a related issue in #239, which @jdhoffa fixed in #239 and released in r2dii.analysis v0.1.2.
* Problem: "unweighted results are coming out wrong when there are multiple loans to the same company" -- @georgeharris2deg
* Expected: "report the projected production values for any given company directly from the ALD." -- @georgeharris2deg

@georgeharris2deg, 

Can you please give more details about the problem and expectation, supporting your comments with a [reprex](https://reprex.tidyverse.org/)? 

What version of r2dii packages are you using? In your reprex please include the version of all r2dii packages or include the session information in your reprex output (see argument `si = TRUE` to `reprex::reprex()`).

``` r
packageVersion("r2dii.data")
#> [1] '0.1.6'
packageVersion("r2dii.match")
#> [1] '0.0.7'
packageVersion("r2dii.analysis")
#> [1] '0.1.3'
```

<sup>Created on 2020-12-22 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>
<details>
<summary>
Session info
</summary>

``` r
devtools::session_info()
#> ─ Session info ───────────────────────────────────────────────────────────────
#>  setting  value                         
#>  version  R version 4.0.3 (2020-10-10)  
#>  os       Ubuntu 18.04.5 LTS            
#>  system   x86_64, linux-gnu             
#>  ui       X11                           
#>  language en_US:en                      
#>  collate  en_US.UTF-8                   
#>  ctype    en_US.UTF-8                   
#>  tz       America/Argentina/Buenos_Aires
#>  date     2020-12-22                    
#> 
#> ─ Packages ───────────────────────────────────────────────────────────────────
#>  package     * version    date       lib source                            
#>  assertthat    0.2.1      2019-03-21 [1] CRAN (R 4.0.3)                    
#>  callr         3.5.1      2020-10-13 [1] CRAN (R 4.0.3)                    
#>  cli           2.2.0      2020-11-20 [1] CRAN (R 4.0.3)                    
#>  crayon        1.3.4      2017-09-16 [1] CRAN (R 4.0.3)                    
#>  desc          1.2.0      2018-05-01 [1] CRAN (R 4.0.0)                    
#>  devtools      2.3.2      2020-09-18 [1] RSPM (R 4.0.2)                    
#>  digest        0.6.27     2020-10-24 [1] CRAN (R 4.0.3)                    
#>  ellipsis      0.3.1      2020-05-15 [1] CRAN (R 4.0.3)                    
#>  evaluate      0.14       2019-05-28 [1] CRAN (R 4.0.0)                    
#>  fansi         0.4.1      2020-01-08 [1] CRAN (R 4.0.3)                    
#>  fs            1.5.0      2020-07-31 [1] RSPM (R 4.0.2)                    
#>  glue          1.4.2      2020-08-27 [1] CRAN (R 4.0.3)                    
#>  highr         0.8        2019-03-20 [1] CRAN (R 4.0.0)                    
#>  htmltools     0.5.0.9003 2020-12-14 [1] Github (rstudio/htmltools@d18bd8e)
#>  knitr         1.30       2020-09-22 [1] RSPM (R 4.0.2)                    
#>  lifecycle     0.2.0      2020-03-06 [1] CRAN (R 4.0.3)                    
#>  magrittr      2.0.1      2020-11-17 [1] CRAN (R 4.0.3)                    
#>  memoise       1.1.0      2017-04-21 [1] CRAN (R 4.0.0)                    
#>  pkgbuild      1.1.0      2020-07-13 [1] RSPM (R 4.0.2)                    
#>  pkgload       1.1.0      2020-05-29 [1] CRAN (R 4.0.0)                    
#>  prettyunits   1.1.1      2020-01-24 [1] CRAN (R 4.0.0)                    
#>  processx      3.4.5      2020-11-30 [1] RSPM (R 4.0.3)                    
#>  ps            1.5.0      2020-12-05 [1] RSPM (R 4.0.3)                    
#>  purrr         0.3.4      2020-04-17 [1] CRAN (R 4.0.0)                    
#>  R6            2.5.0      2020-10-28 [1] RSPM (R 4.0.3)                    
#>  remotes       2.2.0      2020-07-21 [1] RSPM (R 4.0.3)                    
#>  rlang         0.4.9.9000 2020-12-17 [1] Github (r-lib/rlang@bfb1d41)      
#>  rmarkdown     2.5        2020-10-21 [1] RSPM (R 4.0.3)                    
#>  rprojroot     2.0.2      2020-11-15 [1] RSPM (R 4.0.3)                    
#>  sessioninfo   1.1.1      2018-11-05 [1] CRAN (R 4.0.0)                    
#>  stringi       1.5.3      2020-09-09 [1] RSPM (R 4.0.2)                    
#>  stringr       1.4.0      2019-02-10 [1] CRAN (R 4.0.0)                    
#>  testthat      3.0.1      2020-12-17 [1] RSPM (R 4.0.3)                    
#>  usethis       2.0.0.9000 2020-12-11 [1] Github (r-lib/usethis@f96bf2e)    
#>  withr         2.3.0      2020-09-22 [1] RSPM (R 4.0.2)                    
#>  xfun          0.19       2020-10-30 [1] CRAN (R 4.0.3)                    
#>  yaml          2.2.1      2020-02-01 [1] RSPM (R 4.0.0)                    
#> 
#> [1] /home/mauro/R/x86_64-pc-linux-gnu-library/4.0
#> [2] /usr/local/lib/R/site-library
#> [3] /usr/lib/R/site-library
#> [4] /usr/lib/R/library
```

</details>

**Comment by maurolepore** [comment]
For the record, CRAN maintainers are on vacations until Jan 4, 2021. Until then the submission of packages is not possible. But it'd be great to sort this out before then so we can release a patch ASAP.

**Comment by georgeharris2deg** [comment]
I have checked the second issue here Alluding to the corporate_economy appearing in the "name_ald" column. This does not appear to be an issue and was a misunderstanding on my side. 

This row is simply stating that the production values or the corprate_economy as such. I have confirmed that this is being calculated correctly by summing form the ALD 

We can close This part of the above issue. 

However, the main issue still remains that the company level results are being calculated incorrectly when they are multiple loans to the same company.

**Comment by georgeharris2deg** [comment]
@maurolepore reprexed attached below as discussed
Thanks!

``` r
library(tidyverse)
#> Warning: package 'tidyverse' was built under R version 4.0.3
library(r2dii.data)
#> Warning: package 'r2dii.data' was built under R version 4.0.3
library(r2dii.match)
#> Warning: package 'r2dii.match' was built under R version 4.0.3
library(r2dii.analysis)
#> Warning: package 'r2dii.analysis' was built under R version 4.0.3
packageVersion("r2dii.data")
#> [1] '0.1.6'
packageVersion("r2dii.match")
#> [1] '0.0.7'
packageVersion("r2dii.analysis")
#> [1] '0.1.3'

#loanbook with multiple loans to one company 

lbk <- tibble::tribble(
     ~id_loan, ~id_direct_loantaker, ~name_direct_loantaker, ~id_intermediate_parent_1, ~name_intermediate_parent_1, ~id_ultimate_parent, ~name_ultimate_parent, ~loan_size_outstanding, ~loan_size_outstanding_currency, ~loan_size_credit_limit, ~loan_size_credit_limit_currency, ~sector_classification_system, ~sector_classification_input_type, ~sector_classification_direct_loantaker, ~fi_type, ~flag_project_finance_loan, ~name_project, ~lei_direct_loantaker, ~isin_direct_loantaker,
         "L1",               "DL1",              "company A",                        NA,                          NA,              "UP16",             "company A",                 225626,                           "EUR",                18968806,                            "EUR",                        "NACE",                            "Code",                                    3511,   "Loan",                       "No",            NA,                    NA,                     NA,
         "L2",               "DL1",              "company A",                        NA,                          NA,              "UP16",             "company A",                   4321,                           "EUR",                   44333,                            "EUR",                        "NACE",                            "Code",                                    3511,   "Loan",                       "No",            NA,                    NA,                     NA,
     )

# ald of one company 

ald <- tibble::tribble(
    ~name_company, ~sector, ~technology, ~year, ~production, ~production_unit, ~emission_factor, ~ald_emission_factor_unit, ~plant_location, ~is_ultimate_owner, ~ald_timestamp,
        "company A", "power",   "coalcap",  2020,         50,             "MW",               NA,                        NA,            "IT",               TRUE,       "2019Q4"
    )

# load scenerio file

scenario <- r2dii.data::scenario_demo_2020
scenario <- scenario %>% mutate(scenario_source = "weo_2019")

# load region file

region <- r2dii.data::region_isos

# match ALD to lbk

match_file <- match_name(lbk,ald)

# prooritize matches 

lbk_ready <- prioritize(match_file)

# run company level results 

company_results <- target_market_share(lbk_ready, ald, scenario, region, by_company = TRUE, weight_production = FALSE)

# expected value for company level results should be the total production present in the ad for a given company 

expected <- ald$production

# the actual value is outputed in the company level results output and recorded as the metric = projected with all the respective filters  

actual <- company_results %>% filter(
  region == "global", 
  scenario_source == "weo_2019",
  technology == "coalcap",
  metric == "projected"
)

# logical test to see is expected = actual 

identical(actual$production, expected)
#> [1] FALSE
```

<sup>Created on 2020-12-23 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>

**Comment by maurolepore** [comment]
@georgeharris2deg, #256 fixes this issue and is ready for your review [here](https://github.com/2DegreesInvesting/r2dii.analysis/pull/256/files). It's also pending @jdhoffa review but you (or bank-folks) may install this PR with:

```r
remotes::install_github(
  "2DegreesInvesting/r2dii.analysis@253_multi-laon-by-company-yields-unweighted-prod-as-ald"
)
```

--

Your reprex was very useful and help me to capture the essence of the problem in tests/testthat/test-target_market_share.R (line 588 onwards). Here is your same reprex, now returning `TRUE`.

``` r
library(tidyverse)
library(r2dii.data)
library(r2dii.match)
library(r2dii.analysis)
packageVersion("r2dii.data")
#> [1] '0.1.6'
packageVersion("r2dii.match")
#> [1] '0.0.8'
packageVersion("r2dii.analysis")
#> [1] '0.1.3.9000'

#loanbook with multiple loans to one company 

lbk <- tibble::tribble(
     ~id_loan, ~id_direct_loantaker, ~name_direct_loantaker, ~id_intermediate_parent_1, ~name_intermediate_parent_1, ~id_ultimate_parent, ~name_ultimate_parent, ~loan_size_outstanding, ~loan_size_outstanding_currency, ~loan_size_credit_limit, ~loan_size_credit_limit_currency, ~sector_classification_system, ~sector_classification_input_type, ~sector_classification_direct_loantaker, ~fi_type, ~flag_project_finance_loan, ~name_project, ~lei_direct_loantaker, ~isin_direct_loantaker,
         "L1",               "DL1",              "company A",                        NA,                          NA,              "UP16",             "company A",                 225626,                           "EUR",                18968806,                            "EUR",                        "NACE",                            "Code",                                    3511,   "Loan",                       "No",            NA,                    NA,                     NA,
         "L2",               "DL1",              "company A",                        NA,                          NA,              "UP16",             "company A",                   4321,                           "EUR",                   44333,                            "EUR",                        "NACE",                            "Code",                                    3511,   "Loan",                       "No",            NA,                    NA,                     NA,
     )

# ald of one company 

ald <- tibble::tribble(
    ~name_company, ~sector, ~technology, ~year, ~production, ~production_unit, ~emission_factor, ~ald_emission_factor_unit, ~plant_location, ~is_ultimate_owner, ~ald_timestamp,
        "company A", "power",   "coalcap",  2020,         50,             "MW",               NA,                        NA,            "IT",               TRUE,       "2019Q4"
    )

# load scenerio file

scenario <- r2dii.data::scenario_demo_2020
scenario <- scenario %>% mutate(scenario_source = "weo_2019")

# load region file

region <- r2dii.data::region_isos

# match ALD to lbk

match_file <- match_name(lbk,ald)

# prooritize matches 

lbk_ready <- prioritize(match_file)

# run company level results 

company_results <- target_market_share(lbk_ready, ald, scenario, region, by_company = TRUE, weight_production = FALSE)

# expected value for company level results should be the total production present in the ad for a given company 

expected <- ald$production

# the actual value is outputed in the company level results output and recorded as the metric = projected with all the respective filters  

actual <- company_results %>% filter(
  region == "global", 
  scenario_source == "weo_2019",
  technology == "coalcap",
  metric == "projected"
)

# logical test to see is expected = actual 

identical(actual$production, expected)
#> [1] TRUE
```

<sup>Created on 2020-12-24 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>

**Comment by jdhoffa** [comment]
Closed by #256
